python/marvin/contrib/vacs/firefly.py

```diff
@@ -58,15 +74,30 @@ def get_target(self, parent_object):
         plateifu = parent_object.plateifu
         imagesz = int(parent_object.header['NAXIS1'])
 
-        # download the vac from the SAS if it does not already exist locally
-        if not self.file_exists(self.summary_file):
-            log.info('Warning: This file is ~6 GB.  It may take awhile to download')
-            self.summary_file = self.download_vac('mangaffly', path_params=self.path_params)
-
-        # create container for more complex return data.
-        ffly = FFlyTarget(plateifu, vacfile=self.summary_file, imagesz=imagesz)
+        if len(self.summary_file)==1:   # for data releases <DR17
+            # download the vac from the SAS if it does not already exist locally
+            if not self.file_exists(self.summary_file):
+                log.info('Warning: This file is ~6 GB.  It may take awhile to download')
+                self.summary_file = self.download_vac('mangaffly', path_params=self.path_params)
+                
+            # create container for more complex return data.
+            ffly = FFlyTarget(plateifu, vacfile=self.summary_file,imagesz=imagesz)
+            return ffly
+
+        # for DR17 return dictionary; vacfile is only downloaded when key is selected
+        ffly = CallableDict({
+			"miles":lambda: self.prepare_container(plateifu, imagesz, 0),
+			"mastar":lambda: self.prepare_container(plateifu, imagesz, 1)
+            })
 
         return ffly
+    
+    def prepare_container(self,plateifu,imagesz,n):
```

Comment:
can you add a small docstring here describing what this method does?

Reply:
done